src/org/zaproxy/zap/extension/jsonView/HttpPanelJsonView.java

```diff
@@ -0,0 +1,177 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.jsonView;
+
+import java.awt.BorderLayout;
+
+import javax.swing.*;
+
+import net.sf.json.*;
+import org.apache.commons.configuration.FileConfiguration;
+import org.fife.ui.rtextarea.RTextScrollPane;
+import org.parosproxy.paros.network.HttpBody;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.httppanel.Message;
+import org.zaproxy.zap.extension.httppanel.view.*;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.request.RequestBodyStringHttpPanelViewModel;
+import org.zaproxy.zap.extension.httppanel.view.impl.models.http.response.ResponseBodyStringHttpPanelViewModel;
+
+public class HttpPanelJsonView implements HttpPanelView, HttpPanelViewModelListener {
+
+	/**
+	 * Default name used for {@code MessageContainer}.
+	 *
+	 * @see org.zaproxy.zap.view.messagecontainer.MessageContainer
+	 */
+	public static final String DEFAULT_MESSAGE_CONTAINER_NAME = "HttpMessagePanel";
+	public static final String NAME = "HttpPanelJsonView";
+	private static final String CAPTION_NAME = "Json";
+
+	private JTextArea httpPanelJsonArea;
+	private JPanel mainPanel;
+
+	private AbstractStringHttpPanelViewModel model;
+
+	public HttpPanelJsonView(AbstractStringHttpPanelViewModel model) {
+		//TODO disable line numbers, enable folding (requires rsyntaxtextview 2.6.1)
+		httpPanelJsonArea = new HttpPanelJsonArea();
+		RTextScrollPane scrollPane = new RTextScrollPane(httpPanelJsonArea);
+		scrollPane.setLineNumbersEnabled(false);
+		mainPanel = new JPanel(new BorderLayout());
+		mainPanel.add(scrollPane, BorderLayout.CENTER);
+		this.model = model;
+		model.addHttpPanelViewModelListener(this);
+	}
+
+	@Override
+	public void setSelected(boolean selected) {
+		if (selected) {
+			httpPanelJsonArea.requestFocusInWindow();
+		}
+	}
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	@Override
+	public String getCaptionName() {
+		return CAPTION_NAME;
+	}
+
+	@Override
+	public String getTargetViewName() {
+		return "";
+	}
+
+	@Override
+	public int getPosition() {
+		return 0;
+	}
+
+	@Override
+	public boolean isEnabled(Message message) {
+		if(message == null){
+			return false;
+		}
+		String jsonString = null;
+		if(message instanceof HttpMessage) {
+			HttpMessage httpMessage = ((HttpMessage) message);
+			if (this.model instanceof RequestBodyStringHttpPanelViewModel) {
+				jsonString = httpMessage.getRequestBody().toString();
+			} else if (this.model instanceof ResponseBodyStringHttpPanelViewModel) {
+				jsonString = httpMessage.getResponseBody().toString();
+			} else {
+				return false;
+			}
+		}
+		try{
+			JSONFromString(jsonString);

```

Comment:
Enable the view for any string model that parses as valid json. The overhead should be small since the parsing fails fast if the content isn't json. content types are often not reliable, so this enables the views for where it is useful, not just where it should be.